libdnf/sack/query.cpp

```diff
@@ -1915,11 +1925,11 @@ Query::Impl::filterAdvisory(const Filter & f, Map *m, int keyname)
             }
 
             Id id = -1;
-            // Add to candidates resultPset pkgs that match name and arch with some already installed pkg
+            // Add to candidates resultPset pkgs that match name and arch with some already installed pkg or match name and either the installed or candidate are NOARCH
             while ((id = resultPset->next(id)) != -1) {
                 Solvable * s = pool_id2solvable(pool, id);
-                auto low = std::lower_bound(installed_solvables.begin(), installed_solvables.end(), s, NameArchSolvableComparator);
-                if (low != installed_solvables.end() && s->name == (*low)->name && s->arch == (*low)->arch) {
+                auto low = std::lower_bound(installed_solvables.begin(), installed_solvables.end(), s, NameSolvableComparator);
```

Comment:
[minor] It is not an issue but line 1906 (`std::sort(installed_solvables.begin(), installed_solvables.end(), NameArchSolvableComparator);
`) can use only the name comparison key. May be it will provide slightly better performance.